core/src/main/java/org/apache/accumulo/core/spi/scan/ScanServerSelector.java

```diff
@@ -44,139 +37,23 @@ public interface ScanServerSelector {
+  default void init(ScanServerSelectorInitParameters params) {
     Preconditions.checkArgument(params.getOptions().isEmpty(), "No options expected");
   }
 
   /**
-   * this object is used to communicate what the previous actions were attempted, when they were
-   * attempted, and the result of the attempt
-   *
-   * @since 2.1.0
-   */
-  interface ScanAttempt {
-
-    // represents reasons that previous attempts to scan failed
-    enum Result {
-      BUSY, ERROR
-    }
-
-    String getServer();
-
-    long getEndTime();
-
-    Result getResult();
-  }
-
-  /**
-   * This interface exists so that is easier to evolve what is passed to
-   * {@link #determineActions(SelectorParameters)} without having to make breaking changes.
-   *
-   * @since 2.1.0
-   */
-  public interface SelectorParameters {
-
-    /**
-     * @return the set of tablets to be scanned
-     */
-    Collection<TabletId> getTablets();
-
-    /**
-     * @return scan attempt information for the tablet
-     */
-    Collection<? extends ScanAttempt> getAttempts(TabletId tabletId);
-
-    /**
-     * @return any hints set on a scanner using
-     *         {@link org.apache.accumulo.core.client.ScannerBase#setExecutionHints(Map)}. If none
-     *         were set an empty map is returned.
-     */
-    Map<String,String> getHints();
-  }
-
-  public interface Actions {
-
-    /**
-     * @return what scan server to use for a given tablet. Returning null indicates the tablet
-     *         server should be used for this tablet.
-     */
-    String getScanServer(TabletId tabletId);
-
-    /**
-     * @return The amount of time to wait on the client side before starting to contact servers.
-     *         Return {@link Duration#ZERO} if no client side wait is desired.
-     */
-    public Duration getDelay();
-
-    /**
-     * @return The amount of time to wait for a scan to start on the server side before reporting
-     *         busy. For example if a scan request is sent to scan server with a busy timeout of
-     *         50ms and the scan has not started running within that time then the scan server will
-     *         not ever run the scan and it will report back busy. If the scan starts running, then
-     *         it will never report back busy. Setting a busy timeout that is &le; 0 means that it
-     *         will wait indefinitely on the server side for the task to start.
-     */
-    public Duration getBusyTimeout();
-  }
-
-  /**
-   * Uses the SelectorParameters to determine which, if any, ScanServer should be used for scanning
-   * a tablet.
+   * Uses the {@link ScanServerSelectorParameters} to determine which, if any, ScanServer should be
+   * used for scanning a tablet.
    *
    * @param params
    *          parameters for the calculation
    * @return results
    */
-  Actions determineActions(SelectorParameters params);
+  ScanServerSelectorActions determineActions(ScanServerSelectorParameters params);
```

Comment:
In addition to the init parameters, I think it would a be nice to keep the parameters interface for this method nested.

Unrelated to the PR, but looking at this after not having looked at it for a while I think this method needs a different name.  Maybe something like `selectServers` instead of `determineActions` and the return type could be named `ScanServerSelections`.

Reply:
I'll put the init parameters and other parameters back into a nested interface. Do you think that your rename suggestion could be done in this PR as well or should that be done in its own PR?

Reply:
> Do you think that your rename suggestion could be done in this PR as well or should that be done in its own PR?

Seems fine as long as it does not slow down achieving the original intent.

Reply:
Moved the parameters back to nested interfaces in 3d850ae

Reply:
> I think this method needs a different name. Maybe something like `selectServers` instead of `determineActions` and the return type could be named `ScanServerSelections`.

Addressed in e187cd5